### DISCUSSION TITLE
Remote ROS2 connection over LTE

### DISCUSSION THREAD

**Discussion by Jackogip** [foxglove/community#16] [Q&A]
Nice work with foxglove.

Are there any examples of connecting a ROS2 robot with Foxglove over a 4G/LTE connection?

If not, what are the key things to be aware of?

**Comment by jtbandes** [comment]
We have found that the complexity of properly setting up networking for ROS 2 is quite difficult for most applications. I would recommend trying the [rosbridge](https://foxglove.dev/docs/studio/connection/rosbridge) connection, which only requires a single host/port to be accessible for the WebSocket connection. Caveats are:
- The WebSocket protocol is built on TCP, so it depends on a good connection quality and will struggle if packets are being dropped.
- The server does not have any sort of authentication built in, so you are responsible for ensuring the connection is secure.
